### PR TITLE
Add tests to .glotter.yml

### LINKED ISSUE #2930: Add tests to .glotter.yml
I am planning on adding automated test generation using the `.glotter.yml` in glotter2 0.5.0

### PR DESCRIPTION
**Pull request by rzuckerm**
The tests are specified in the `tests` item for each project. Transformations to be done to the test outputs are specified in the `transformations` for each project. Since all the project tests are auto-generated, all the existing test files have been deleted. The auto-generated test files can be viewed [here](https://github.com/rzuckerm/sample-programs-auto-gen-tests/tree/main/test/generated).

fixes #2930

### CODE REVIEW

File: .glotter.yml [hunk 1]
```diff
@@ -8,286 +8,295 @@ projects:
...
+                    -   name: "sample input one false"
+                        input: '"5" "7"'
+                        expected: "false"
+                    -   name: "sample input many false"
+                        input: '"1, 3, 5, 6" "7"'
+                        expected: "false"
+                    -   name: "sample input middle true"
+                        input: '"1, 2, 3, 4, 5, 6, 7" "3"'
+                        expected: "true"
+                transformations:
```
Comment (jrg94) [inline .glotter.yml:66]: Love this! Very cool feature. Allows this to be expanded, so we don't worry about missing the old python setup.

**Review by jrg94 — COMMENTED** [review]
Looks great! Do we need some of the files that are floating around root now? Like runner.py? If so, does it still need to be maintained (e.g., add in the enums for each test set)?

**Comment by rzuckerm** [comment]
> Looks great! Do we need some of the files that are floating around root now? Like runner.py? If so, does it still need to be maintained (e.g., add in the enums for each test set)?

Thanks! I removed all the test files and trimmed `runner.py` down to just running `glotter.main()`. Ideally, I'd just like the user to be able to run `glotter` directly, but I don't know how this will work for Windows users. Unfortunately, I cannot try this out on my Windows machine since I run VirtualBox for the Linux environment, and docker for Windows needs to enable a type of virtualization that is incompatible with VirtualBox :frowning_face: . I put a placeholder `ProjectType` class in `test/__init__.py` in case a new project comes along that cannot use the auto-generated test framework.

**Review by jrg94 — APPROVED** [review]
Amazing work! I look forward to this change for everyone who keeps working on this project.

**Pull request merged**